GitFyle.Core.Api.Tests.Unit/Services/Foundations/Configurations/ConfigurationServiceTests.Validations.Add.cs

```diff
@@ -0,0 +1,328 @@
+                key: nameof(Configuration.Name),
+                values: "Text is required.");
+
+            invalidConfigurationException.AddData(
+                key: nameof(Configuration.Value),
+                values: "Text is required.");
+
+            invalidConfigurationException.AddData(
+                key: nameof(Configuration.CreatedBy),
+                values: "Text is required.");
+
+            invalidConfigurationException.AddData(
+                key: nameof(Configuration.CreatedDate),
+                values: "Date is invalid.");
+
+            invalidConfigurationException.AddData(
+                key: nameof(Configuration.UpdatedBy),
+                values: "Text is required.");
+
+            invalidConfigurationException.AddData(
+                key: nameof(Configuration.UpdatedDate),
+                values: "Date is invalid.");
+
+            var expectedConfigurationValidationException =
+                new ConfigurationValidationException(
+                    message: "Configuration validation error occurred, fix the errors and try again.",
+                    innerException: invalidConfigurationException);
+
+            this.datetimeBrokerMock.Setup(broker =>
+                broker.GetCurrentDateTimeOffsetAsync())
+                    .ReturnsAsync(randomDateTimeOffset);
+
+            // when
+            ValueTask<Configuration> addConfigurationTask =
+                this.configurationService.AddConfigurationAsync(invalidConfiguration);
+
+            ConfigurationValidationException actualConfigurationValidationException =
+                await Assert.ThrowsAsync<ConfigurationValidationException>(
+                    addConfigurationTask.AsTask);
+
+            // then
+            actualConfigurationValidationException.Should()
+                .BeEquivalentTo(expectedConfigurationValidationException);
+
+            this.datetimeBrokerMock.Verify(broker =>
+                broker.GetCurrentDateTimeOffsetAsync(),
+                    Times.Once);
+
+            this.loggingBrokerMock.Verify(broker =>
+                broker.LogErrorAsync(It.Is(
+                    SameExceptionAs(expectedConfigurationValidationException))),
+                        Times.Once);
+
+            this.storageBrokerMock.Verify(broker =>
+                broker.InsertConfigurationAsync(It.IsAny<Configuration>()),
+                    Times.Never);
+
+            this.datetimeBrokerMock.VerifyNoOtherCalls();
+            this.loggingBrokerMock.VerifyNoOtherCalls();
+            this.storageBrokerMock.VerifyNoOtherCalls();
+        }
+
+        [Fact]
+        public async Task ShouldThrowValidationExceptionOnAddIfConfigurationHasInvalidLengthPropertiesAndLogItAsync()
+        {
+            // given
+            DateTimeOffset randomDateTimeOffset = GetRandomDateTimeOffset();
+            var invalidConfiguration = CreateRandomConfiguration(dateTimeOffset: randomDateTimeOffset);
```

Comment:
Remove the parameter hint.

Reply:
Fixed.
![image](https://github.com/user-attachments/assets/15923f0a-ab5e-44ee-a23a-e22653d50111)